libafl/src/stages/mutational.rs

```diff
@@ -153,25 +151,22 @@ pub static DEFAULT_MUTATIONAL_MAX_ITERATIONS: usize = 128;
 
 /// The default mutational stage
 #[derive(Clone, Debug)]
-pub struct StdMutationalStage<E, EM, I, M, Z> {
+pub struct StdMutationalStage<I, M> {
     /// The name
     name: Cow<'static, str>,
     /// The mutator(s) to use
     mutator: M,
     /// The maximum amount of iterations we should do each round
     max_iterations: usize,
-    #[allow(clippy::type_complexity)]
-    phantom: PhantomData<(E, EM, I, Z)>,
+    phantom: PhantomData<I>,
 }
 
-impl<E, EM, I, M, Z> MutationalStage<E, EM, I, M, Z> for StdMutationalStage<E, EM, I, M, Z>
+impl<E, EM, I, M, S, Z> MutationalStage<E, EM, I, M, S, Z> for StdMutationalStage<I, M>
 where
-    E: UsesState<State = Self::State>,
-    EM: UsesState<State = Self::State>,
-    M: Mutator<I, Self::State>,
-    Z: Evaluator<E, EM>,
-    Self::State: HasCorpus + HasRand + HasExecutions + HasMetadata + HasNamedMetadata,
-    I: MutatedTransform<Self::Input, Self::State> + Clone,
+    M: Mutator<I, S>,
+    Z: Evaluator<E, EM, I, S>,
+    S: HasCorpus + HasRand + HasExecutions + HasMetadata + HasNamedMetadata,
```

Comment:
Only `S: HasRand` is needed here.

Reply:
`I` bound is still needed too :joy:

Reply:
deleting both of them didn't give any error, so i thought it's not neccesarry

Reply:
That's just because the build doesn't complete. We don't get generic errors until the syntax and type errors are fixed across the whole build. (cc @domenukk this was the issue I mentioned)

Reply:
As I stated earlier, just comment out the mods and bring them back one by one later

Reply:
... I definitely missed that and will now be using this strategy, thanks lol